server/views/WfModule.py

```diff
@@ -247,6 +247,40 @@ def wfmodule_input(request, pk, format=None):
         else:
             return table_result(request, prev_modules.last())
 
+
+# returns a list of columns and their simplified types
+@api_view(['GET'])
+@renderer_classes((JSONRenderer,))
+@permission_classes((IsAuthenticatedOrReadOnly, ))
+def wfmodule_columns(request, pk, format=None):
+    if request.method == 'GET':
+        try:
+            wf_module = WfModule.objects.get(pk=pk)
+        except WfModule.DoesNotExist:
+            return HttpResponseNotFound()
+
+        if not wf_module.workflow.user_authorized_read(request.user):
+            return HttpResponseForbidden()
+
+        table = execute_wfmodule(wf_module)
+        dtypes = table.dtypes.to_dict()
+        ret_types = []
+        for col in dtypes:
+            # We are simplifying the data types here.
+            # More stuff can be added to these lists if we run into anything new.
+            stype = "String"
```

Comment:
This logic should probably go in the usual render() results 
- because we already use a zero-row render to get column names for column parameters (see WorkbenchAPI.inputColumns) so we shouldn't duplicate that functionality here
- so other modules can use returned column types without having to make a separate API call